pymeasure/instruments/agilent/agilentE4407B.py

```diff
@@ -0,0 +1,637 @@
+        in Hz. This property can be set.
+        """,
+        validator=truncated_range,
+        values=[9000, 26500000000],
+        cast=int,
+    )
+
+    span = Instrument.control(
+        ":SENS:FREQ:SPAN?",
+        ":SENS:FREQ:SPAN %g",
+        """ A floating point property that represents the span
+        in Hz. This property can be set.
+        """,
+    )
+
+    def full_span(self):
+        """Sets the span to the full span of the instrument."""
+        self.write(":SENS:FREQ:SPAN:FULL")
+
+    def last_span(self):
+        """
+        A command that sets the span to the previous span.
+        """
+        self.write(":SENS:FREQ:SPAN:PREV")
+
+    # sweep commands
+    frequency_points = Instrument.control(
+        ":SENSe:SWEEp:POINts?",
+        ":SENSe:SWEEp:POINts %g",
+        """ An integer property that represents the number of frequency
+        points in the sweep. This property can take values from 101 to 8192.
+        """,
+        validator=truncated_range,
+        values=[101, 8192],
+        cast=int,
+    )
+    sweep_time = Instrument.control(
+        ":SENS:SWE:TIME?",
+        ":SENS:SWE:TIME %g",
+        """ A floating point property that represents the sweep time
+        in seconds. This property can be set.
+        """,
+    )
+    number_of_segments = Instrument.measurement(
+        ":SENS:SWEep:SEGMent:COUNT?",
+        """ An integer property that represents the number of segments
+        in the sweep. This property is read-only.
+        """,
+    )
+    set_all_segments_sst = Instrument.control(
+        ":SENS:SWE:SEGM:DATA? SST",
+        ":SENS:SWE:SEGM:DATA SST,%g",
+        """ A command that sets all the segments of a sweep, at once, with a string.
+        format is start, stop, rbw, vbw, points, time
+        """,
+    )
+    set_all_segments_csp = Instrument.control(
+        ":SENS:SWE:SEGM:DATA? CSP",
+        ":SENS:SWE:SEGM:DATA CSP,%g",
+        """ A command that sets all the segments of a sweep, at once, with a string.
+        format is center, span, rbw, vbw, points, time
+        """,
+    )
+    merge_segments_sst = Instrument.setting(
+        ":SENS:SWE:SEGM:DATA:MERge SST %g",
+        """
+        A command that merges the data with current the segments of a sweep with a string.
+        format is start, stop, rbw, vbw, points, time
+        """,
+    )
+    merge_segments_csp = Instrument.setting(
+        ":SENS:SWE:SEGM:DATA:MERge CSP %g",
+        """
+        A command that merges the data with current the segments of a sweep with a string.
+        format is center, span, rbw, vbw, points, time
+        """,
+    )
+
+    def delete_segments(self):
+        """Deletes all the segmented sweep data."""
+        self.write(":SENS:SWE:SEGM:DEL")
+
+    delete_segment = Instrument.setting(
+        ":SENS:SWE:SEGM:DEL %g",
+        """
+        Deletes the specifed segment of a sweep.""",
+    )
```

Comment:
This is unusual as a property, you'd "assign" the segment to delete to the property. Better to have another method (but then name the former `delete_all_segments` for clarity), or handle it with an argument to the above method.